app/src/main/java/com/totsp/crossword/BackgroundDownloadService.java

```diff
@@ -0,0 +1,156 @@
+package com.totsp.crossword;
+
+import android.Manifest;
+import android.annotation.TargetApi;
+import android.app.NotificationManager;
+import android.app.job.JobInfo;
+import android.app.job.JobParameters;
+import android.app.job.JobScheduler;
+import android.app.job.JobService;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.os.AsyncTask;
+import android.os.Looper;
+import android.preference.PreferenceManager;
+import android.support.v4.content.ContextCompat;
+
+import com.totsp.crossword.net.Downloaders;
+
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+// Currently only available on API version >=21 due to use of JobScheduler.
+// It may be possible to implement this functionality using AlarmManager for lower SDK versions.
+@TargetApi(21)
+public class BackgroundDownloadService extends JobService {
+    public static final String DOWNLOAD_PENDING_PREFERENCE = "backgroundDlPending";
+
+    private static final Logger LOGGER =
+            Logger.getLogger(BackgroundDownloadService.class.getCanonicalName());
+
+    private static JobInfo getJobInfo(boolean requireUnmetered, boolean allowRoaming,
+                                      boolean requireCharging) {
+        JobInfo.Builder builder = new JobInfo.Builder(
+                JobSchedulerId.BACKGROUND_DOWNLOAD.id(),
+                new ComponentName("com.totsp.crossword.shortyz",
+                        BackgroundDownloadService.class.getName()));
+
+        builder.setPeriodic(TimeUnit.HOURS.toMillis(1))
+                .setRequiredNetworkType(JobInfo.NETWORK_TYPE_UNMETERED)
```

Comment:
I did consider this, but the JobScheduler API doesn't support scheduling at a specific time of day (only periodically or after a given deadline) so scheduling the job consistently at 07:00UTC is non-trivial (the delay needs to change if the user changes time zones, DST, etc.).  We'll also have to implement some handling of failed downloads, whereas now we just converge on the next hourly run.

If your main concern is device battery usage, I've been running this for the past few months and haven't seen Shortyz noticeable in my battery stats (obviously a pretty small sample size).  My guess is JobScheduler batches pretty efficiently.  There's also an option to download only while charging, which we could change to a default.

I also have a personal preference to make this work with PR #88 which would require also scheduling at 04:00UTC to get the NYT puzzle the night before.  I could just hard code two daily runs to make this work for now if we go this route, so not a big deal.

Let me know how strongly you feel about this.  I can spend some time figuring out a way to cut down the number of download attempts if you'd like.